agent/src/beerocks/slave/backhaul_manager/wan_monitor.cpp

```diff
@@ -134,7 +134,9 @@ wan_monitor::ELinkState wan_monitor::process()
             struct ifinfomsg *ifi = (struct ifinfomsg *)NLMSG_DATA(hnl);
 
             // Convert the interface index to name
-            if_indextoname(ifi->ifi_index, name);
+            auto iface_index_name = network_utils::linux_get_iface_name(ifi->ifi_index);
+            std::copy_n(iface_index_name.begin(), IFNAMSIZ - 1, name);
+            name[IFNAMSIZ - 1] = '\0';
```

Comment:
Since you're terminating the string everywhere, I guess it's better to simply have this logic moved into the `network_utils::linux_get_iface_name` method itself.

Reply:
@vitalybu `network_utils::linux_get_iface_name()` returns the C++ string. I think it is a bad idea to add '\0' to C++ string container.

Reply:
True, but first it uses a local variable (i.e. `iface_name`) for storing the output of `if_indextoname`.
You can terminate `iface_name` before returning it as `std::string`...

Reply:
Agree with you, will fix it!

Reply:
@vitalybu fixed!